### ISSUE TITLE
 extract_area: bad extract area

### ISSUE THREAD

**Issue by Alexufo** [lovell/sharp#1907]
If we try to extract more pixels than has  an image - we will get  "extract_area: bad extract area"

Can we ignore exception and extract an actual pixels? I don't want calculate if  image.width < etc.. i suppose it should be in a core. May be additional option?

**Comment by lovell** [comment]
Good idea, we should improve the error handling at the same time to avoid making any calls into libvips that could generate the dread "bad extract area" error.

**Comment by oliviertassinari** [comment]
A detailed error message would also help
